ensembl_metadata/models/genome.py

```diff
@@ -1,160 +1,160 @@
+    is_reference = models.BooleanField(default=False)
 
     class Meta:
-        unique_together = (('genome', 'type', 'name', 'genome_database'),)
+        db_table = 'organism_group_member'
+        unique_together = (('organism', 'organism_group'),)
 
 
-class GenomeAnnotation(models.Model):
-    genome_annotation_id = models.AutoField(primary_key=True)
-    genome = models.ForeignKey(Genome, on_delete=models.CASCADE,
-                               related_name='annotations')
-    type = models.CharField(max_length=32)
-    value = models.CharField(max_length=128)
-    genome_database = models.ForeignKey(GenomeDatabase, on_delete=models.CASCADE,
-                                        related_name='annotations')
+class Genome(models.Model):
+    genome_id = models.AutoField(primary_key=True)
+    genome_uuid = models.CharField(max_length=128, default=uuid.uuid1, unique=True)
+    assembly = models.ForeignKey(Assembly, models.CASCADE, related_name='genomes')
+    organism = models.ForeignKey(Organism, models.CASCADE, related_name='genomes')
+
+    def releases(self):
+        return GenomeRelease.objects.filter(genome_id=self.genome_id)
 
     class Meta:
-        unique_together = (('genome', 'type', 'genome_database'),)
+        db_table = 'genome'
 
 
-class GenomeDivision(models.Model):
-    genome_division_id = models.AutoField(primary_key=True)
-    division = models.ForeignKey(Division, on_delete=models.CASCADE,
-                                 related_name='divisions')
-    genome = models.ForeignKey(Genome, on_delete=models.CASCADE,
-                               related_name='divisions')
+class DatasetSource(models.Model):
+    class SourceType(models.TextChoices):
+        CORE = 'core'
+        CDNA = 'cdna'
+        DATAFILE = 'datafile'
+        OTHERFEATURES = 'otherfeatures'
+        RNASEQ = 'rnaseq'
+        COMPARA = 'compara'
+        FUNCGEN = 'funcgen'
+        VARIATION = 'variation'
 
+    dataset_source_id = models.AutoField(primary_key=True)
+    type = models.CharField(max_length=32, choices=SourceType.choices)
+    name = models.CharField(max_length=255, unique=True)
 
-class GenomeEvent(models.Model):
-    genome_event_id = models.AutoField(primary_key=True)
-    genome = models.ForeignKey(Genome, on_delete=models.CASCADE,
-                               related_name='events')
-    type = models.CharField(max_length=32)
-    source = models.CharField(max_length=128, blank=True, null=True)
-    creation_time = models.DateTimeField(auto_now_add=True)
-    details = models.TextField(blank=True, null=True)
+    class Meta:
+        db_table = 'dataset_source'
+
+
+class DatasetType(models.Model):
+    class DatasetTopic(models.TextChoices):
+        ASSEMBLY = 'assembly'
+        GENESET = 'geneset'
+        ASSEMBLY_ANNOTATION = 'assembly_annotation'
+        GENESET_ANNOTATION = 'geneset_annotation'
+        COMPARATIVE = 'comparative'
+        REGULATION = 'regulation'
+        VARIATION = 'variation'
```

Comment:
From CDM meeting, I think we should add a topic for XRef, unless your plan was to associate such things as "RefSeq" links to geneset_annotation. In such case, I don't know where we cold store this "reationship"

Reply:
I'd intended these as quite high-level groupings; cross-references would indeed be `geneset_annotation`s. My thinking was that we would have maybe two or three different xref dataset types (e.g. 'checksum_based_xref', 'alignment_based_xref'), with GO terms and protein domains also coming under this topic. I wasn't thinking that a dataset would be at the level of a xref source, e.g. 'refseq_xref', 'uniparc_xref' - that seems too granular. What we could/should do is have dataset_statistics with metadata that reflects the xref annotation, to reflect the provenance of the various source dbs, e.g. 'RefSeq count', 'RefSeq version'.